ush/templates/FV3LAM_wflow.xml

```diff
@@ -1200,21 +1364,68 @@ MODULES_RUN_TASK_FP script.
 ************************************************************************
 ************************************************************************
 -->
-  <task name="&ARCHIVE_TN;" cycledefs="archive" maxtries="{{ maxtries_run_post }}">
+  <task name="&ARCHIVE_TN;{{ uscore_ensmem_name }}" cycledefs="archive" maxtries="{{ maxtries_run_post }}">
 
     &RSRV_HPSS;
 
     <command>&JOBSDIR;/../scripts/exregional_archive.ksh</command>
     <cores>1</cores>
     <walltime>04:00:00</walltime>
     <memory>24G</memory>
-    <jobname>&TAG;_&ARCHIVE_TN;</jobname>
-    <join><cyclestr>&LOGDIR;/&ARCHIVE_TN;_@Y@m@d@H.log</cyclestr></join>
+    <jobname>&TAG;_&ARCHIVE_TN;{{ uscore_ensmem_name }}</jobname>
+    <join><cyclestr>&LOGDIR;/&ARCHIVE_TN;{{ uscore_ensmem_name }}_@Y@m@d@H.log</cyclestr></join>
 
     <envar><name>GLOBAL_VAR_DEFNS_FP</name><value>&GLOBAL_VAR_DEFNS_FP;</value></envar>
 
   </task>
 
 {%- endif %}
 
+{%- if do_ensemble %}
+  </metatask>
+{%- endif %}
+
+{%- if do_gsiobserver %}
+<!--
+************************************************************************
+************************************************************************
+-->
+<metatask name="gsi_observer"> 
+<var name="fhr"> {% for h in range(1, 2) %}{{ " %02d" % h  }}{% endfor %} </var>
+   <task name="&ANAL_GSI_OBSERVER_TN;_f#fhr#" cycledefs="#type#cyc" maxtries="{{ maxtries_gsi_observer }}">
+
+     &RSRV_DEFAULT;
+     &WALL_LIMIT_ANAL;  
+
+     <command>&LOAD_MODULES_RUN_TASK_FP; "&RUN_ANAL_TN;" "&JOBSDIR;/JREGIONAL_RUN_OBSERVER"</command>
+     <cores>240</cores>
+     <native> --cpus-per-task 2 --exclusive </native>
+     <walltime>00:40:00</walltime>
+     <jobname>rrfse_&ANAL_GSI_OBSERVER_TN;_f#fhr#</jobname>
+     <join><cyclestr>&LOGDIR;/&ANAL_GSI_OBSERVER_TN;_f#fhr#_@Y@m@d@H.log</cyclestr></join>
+   
+     <envar><name>GLOBAL_VAR_DEFNS_FP</name><value>&GLOBAL_VAR_DEFNS_FP;</value></envar>
+     <envar><name>PDY</name><value><cyclestr>@Y@m@d</cyclestr></value></envar>
+     <envar><name>CDATE</name><value><cyclestr>@Y@m@d@H</cyclestr></value></envar>
+     <envar><name>ANAL_DATE</name><value><cyclestr offset="#fhr#:00:00">@Y@m@d@H</cyclestr></value></envar>
+     <envar><name>CYCLE_DIR</name><value><cyclestr>&CYCLE_BASEDIR;/@Y@m@d@H</cyclestr></value></envar>
+     <envar><name>FG_ROOT</name><value><cyclestr>&CYCLE_BASEDIR;</cyclestr></value></envar>
+     <envar><name>BKTYPE</name><value><cyclestr>0</cyclestr></value></envar>
+     <envar><name>NUM_ENS_MEMBERS</name><value><cyclestr>#{{ num_ens_members }}#</cyclestr></value></envar>
+
+     <dependency>
+       <and>   
+         <datadep age="05:00"><cyclestr offset="#fhr#:00:00">&OBSPATH;/@Y@m@d@H.rap.t@Hz.prepbufr.tm00</cyclestr></datadep>
```

Comment:
A comment on the dependency on prepbufr file. The anal_gsi job does not depend on this. 
So if for some reason, prebufr is missing, the GSI observer will not run, right?

Reply:
Yes, if no prepbufr data available, running the observer makes no sense.

Reply:
ok, thanks.  @hu5970  How do you handle this for anal_gsi jobs? I just feel it is some kind of odd that anal_gsi does not have this prepbufr dependency? It is handled in another task?